### PR TITLE
[Batch mode]: Use first primary name for batch input file name.

### PR DESCRIPTION
**Pull request by davidungar**

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

File: lib/Frontend/FrontendInputsAndOutputs.cpp [hunk 1]
```diff
@@ -124,12 +124,11 @@ FrontendInputsAndOutputs::getRequiredUniquePrimaryInput() const {
...
 std::string FrontendInputsAndOutputs::getStatsFileMangledInputName() const {
-  // FIXME: "batch" should probably be some concatenation of all the primary
-  // input names, in order to keep the stats file names unique. (Or perhaps just
-  // the first primary?)
+  // Use the first primary, even if there are multiple primaries.
+  // That's enough to keep the file names unique.
   return isWholeModule()
              ? "all"
-             : primaryInputCount() == 1 ? firstPrimaryInput().file() : "batch";
+             : hasPrimaryInputs() ? firstPrimaryInput().file() : "batch";
```
Comment (jrose-apple) [inline lib/Frontend/FrontendInputsAndOutputs.cpp]: We think the else branch is dead now, right?

**Review by graydon — APPROVED** [review]
Thinking this over, I'm not even sure this matters; the filenames are made-unique two separate additional ways (by timestamp and by random suffix) so I think even just leaving "batch" would be ok. And might be less-confusing? But either way, it really doesn't matter, we always aggregate stats by module at the finest.

**Comment by jrose-apple** [comment]
Ah, I forgot they were suffixed. I still like being able to identify which one came from which batch, though.

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
Let me know what the two of you decide: batch, first, concatenated with what? And I'll do it. Tx.

**Comment by graydon** [comment]
@davidungar Go with Jordan's preference, it's strictly more information and the tooling shouldn't care.

**Comment by davidungar** [comment]
@swift-ci please smoke test

File: lib/Frontend/FrontendInputsAndOutputs.cpp (hunk 1 above)
Comment (davidungar) [reply lib/Frontend/FrontendInputsAndOutputs.cpp]: Thanks! I've fixed it.

**Comment by davidungar** [comment]
@jrose-apple  I have eliminated the dead case. Let me know if you think of anything else this needs, tx!

**Review by jrose-apple — APPROVED** [review]
Thanks!

**Comment by davidungar** [comment]
My pleasure.

**Pull request merged**